primitives/runtime/src/lib.rs

```diff
@@ -479,7 +482,7 @@ pub enum DispatchError {
 		/// Module index, matching the metadata module index.
 		index: u8,
 		/// Module specific error value.
-		error: u8,
+		error: [u8; MAX_PALLET_ERROR_ENCODED_SIZE],
```

Comment:
Why is this an array? Why not a Compact for example? Or a vec?

Reply:
Vec is a dumb idea actually, but Compact would be nice here.

Reply:
Then we could also remove the `MAX_PALLET_ERROR_ENCODED_SIZE` constant.

Reply:
The `DispatchError` derives `Copy`, so I needed a type that also implements `Copy`. I don't know what the consequences are for removing the `Copy` derivation, so I didn't want to change it into a `Vec` or a `Compact`.

Reply:
Hmm `Copy`. Can you see what happens when you remove it? Otherwise you could also add copy to `Compact` if the inner type implements `Copy`.

Reply:
That would make `DispatchErrorWithPostInfo` also not implement `Copy`, and that's what I would be a bit worried about, since this type is already public and is being used downstream by many users, although... we do have a history of breaking APIs so that may not really be a huge problem. I'd probably still prefer adding a `Copy` impl for `Compact` if the inner type is `Copy` though.

But more to the point: what does `Compact` really help with here? Do I put `Compact<u64>` or something similar?